src/DatePicker/DatePicker.js

```diff
@@ -303,6 +308,8 @@ class DatePicker extends Component {
                             <input
                                 {...inputProps}
                                 className={datePickerInputClasses}
+                                onBlur={() => onBlur({ date: this.state.selectedDate,
+                                    formattedDate: this.state.formattedDate })}
```

Comment:
This would be better written as a separate function in the component and then referenced here like `onBlur={this._handleBlur}`.  Otherwise, this actually creates a new instance of the function each time it goes through the `render` function.